price-feeder/oracle/provider/huobi.go

```diff
@@ -27,140 +32,248 @@ type (
 	//
 	// REF: https://huobiapi.github.io/docs/spot/v1/en/#market-data
 	HuobiProvider struct {
-		baseURL string
-		client  *http.Client
-	}
-
-	// HuobiTraceResponse defines the response type for the last executed trade
-	// for a given ticker/symbol.
-	HuobiTraceResponse struct {
-		Status string `json:"status"`
-		ErrMsg string `json:"err-msg"`
-		Tick   struct {
-			Data []struct {
-				Price float64 `json:"price"`
-			} `json:"data"`
-		} `json:"tick"`
-	}
-
-	// HuobiMarketResponse defines the response type for the last 24h market summary
-	// for a given ticker/symbol.
-	HuobiMarketResponse struct {
-		Status string `json:"status"`
-		ErrMsg string `json:"err-msg"`
-		Tick   struct {
-			Vol float64 `json:"vol"`
-		} `json:"tick"`
+		wsURL           url.URL
+		wsClient        *websocket.Conn
+		logger          zerolog.Logger
+		mu              sync.Mutex
+		tickers         map[string]HuobiTicker // market.$symbol.ticker => HuobiTicker
+		subscribedPairs []types.CurrencyPair
+		reconnectTicker *time.Ticker
```

Comment:
same as the previous PR -- this doesn't really have a need to be a part of the `HuobiProvider` type.

Reply:
done at https://github.com/umee-network/umee/pull/569/commits/576f19d2c689b558508636c8419600b5c3c75aaf